spec.bs

```diff
@@ -221,6 +224,35 @@ The <dfn for="interest group">estimated size</dfn> of an [=interest group=] |ig|
     [=interest group ad/render url=].
   1. The [=string/length=] of |ad|'s [=interest group ad/metadata=] if the field is not null.
 
+<div algorithm="check interest group permissions">
+
+To <dfn>check interest group permissions</dfn> given a [=string=] <dfn
+for="fetch interest group permissions">|ownerOrigin|</dfn>, a [=string=] <dfn
+for="fetch interest group permissions">|frameOrigin|</dfn>, and a [=boolean=]
+<dfn for="fetch interest group permissions">|isJoin|</dfn>:
+1. If |ownerOrigin| is equal to |frameOrigin|, then return true.
+1. Let |permissionsUrl| be the result of [building an interest group permissions url=]
+  with |ownerOrigin| and |frameOrigin|.
+1. Let |resource| be the result of [=fetching resource=] with |permissionsUrl|
+  and "`application/json`". (TODO: use cors mode)
+1. If |resource| is failure, then return false.
+1. Let |permissions| be the result of parsing |resource| as JSON, returning false on failure.
+1. If |isJoin| is true and |permissions| is a JSON object containing key
+  "`joinAdInterestGroup`" with value `true`, return true.
+1. If |isJoin| is false and |permissions| is a JSON object containing key
+  "`leaveAdInterestGroup`" with value `true`, return true.
+1. Return false.
+
+</div>
+
+<div algorithm="build an interest group permissions url">
+
+To <dfn>build an interest group permissions url</dfn> given a [=string=] |ownerOrigin| and a [=string=] |frameOrigin|:
+1. Return the string formed by [=string/concatenating=] "`https://`", |ownerOrigin|,

```

Comment:
You have to serialize the origin as a string: https://html.spec.whatwg.org/multipage/browsers.html#ascii-serialisation-of-an-origin. This should get you out of manually adding `https://` anyways